### DISCUSSION TITLE
storing data structures as oposed to a decorator single object as root

### DISCUSSION THREAD

**Discussion by guykatz** [microstream-one/microstream#534] [Q&A] [answered]
Hi guys;
is there a best practive as for what to save as the root object?
I already know I can store a java collection at the root but is it a good practice? should I wrap the collection in a single object and then store it?
thanks

**Comment by hg-ms** [comment, marked as answer]
Hello,
You can store (nearly) any kind of object as root object. Using a custom object instead of a collection as root is always a good option. That object may be composed of several collections, values and references to other objects.
The main benefit is that you can easily add new members to the custom root object class later on. 

Edit:
I almost forgot the most obvious point:
With such an custom root object you can structure your data in a more “natural” way. E.g., if you’re working on a shop, you can create a class “MyShop” as root that has Customers, Products, etc.
